src/decode.h

```diff
@@ -236,6 +236,19 @@ struct rv_insn_t {
 #if RV32_HAS(EXT_C)
     uint8_t shamt;
 #endif
+
+    /* instruction length */
+    uint8_t insn_len;
+};
+
+/* a translated basic block */
+struct block_t {
```

Comment:
The suffix `_t` implies the use of typedef. Thus, you shall rewrite as following:
```c
typedef struct {
   ...
} block_t;
```